azavea/cac-tripplanner
# Upgrade to Django 3.2

## Overview

Gets us onto the most recent long-term support release.


### Demo

Not much to demo -- everything should remain the same.


### Notes

The app actually worked mostly fine on 3.2 without additional work, probably because anything deprecated in 3.2 had already been addressed by earlier rounds of upgrade work. I tried to do the same thing -- I addressed as many deprecation warnings as I could, even though they were mostly for Django 4+.

Many of the issues I encountered stemmed from `django-wpadmin`, which is a Django admin theme that attempts to replicate the look and feel of the WordPress admin. The upstream package is no longer maintained, so we've long since forked it, and I continued updating our fork.

## Testing Instructions

 * I assume it's because we're installing `django-wpadmin` from a GitHub repo rather than PyPi, but reprovisioning didn't pick up the changes to `requirements.txt`. Instead, I had to do the following to get it to install:
   * `vagrant up && vagrant ssh app`
   * `sudo pip3 uninstall django-wpadmin` (and then logout)
   * `vagrant provision app`
 * Next, run tests and the dev server to make sure that you don't see any deprecation warnings from Django (there will be a couple from boto):
   * `vagrant ssh app`
   * `cd /opt/app/python/cac_tripplanner`
   * `python3 -Wall manage.py test`
   * `sudo systemctl stop cac-tripplanner-app` (This stops the server process that comes up by default when you start the VM; you need to do this before starting up the Django dev server manually because you'll get a port conflict if you don't)
   * `python3 -Wall manage.py runserver`
 * Last, click around the app and make sure everything works as it used to.
 * Pay special attention to the admin site -- I encountered quite a few CSS issues that caused the layout to get messed up, so it's very possible that there are still some more lingering that I've missed.


## Checklist


Closes #1265
